pkg/sinks/clickhouse.go

```diff
@@ -0,0 +1,250 @@
+package sinks
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/resmoio/kubernetes-event-exporter/pkg/kube"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type ClickHouseConfig struct {
+	Hosts             []string `yaml:"hosts"`
+	Username          string   `yaml:"username"`
+	Password          string   `yaml:"password"`
+	Database          string   `yaml:"database"`
+	TableName         string   `yaml:"tableName"`
+	UseHttp           bool     `yaml:"useHttp"`
+	TLS               *TLS     `yaml:"tls"`
+	CreateTable       bool     `yaml:"createTable"`
+	Compress          string   `yaml:"compress"`
+	MaxIdleConns      *int     `yaml:"maxIdleConns"`
+	MaxOpenConns      *int     `yaml:"maxOpenConns"`
+	ConnMaxLifetimeMs *int     `yaml:"connMaxLifetimeMs"`
+	ConnMaxIdleTimeMs *int     `yaml:"connMaxIdleTimeMs"`
+}
+
+const (
+	ddlFmt = `CREATE TABLE %s (
+	KubeClusterName LowCardinality(String),
+	Reason LowCardinality(String),
+	Message String,
+	SourceComponent LowCardinality(Nullable(String)),
+	SourceHost Nullable(String),
+	FirstTimestamp Nullable(DateTime),
+	LastTimestamp Nullable(DateTime),
+	Count Nullable(UInt32),
+	Type LowCardinality(Nullable(String)),
+	EventTime Nullable(DateTime),
+	InvolvedObjectKind LowCardinality(Nullable(String)),
+	InvolvedObjectNamespace LowCardinality(Nullable(String)),
+	InvolvedObjectName LowCardinality(Nullable(String)),
+	InvolvedObjectUID Nullable(UUID),
+	InvolvedObjectResourceVersion Nullable(String),
+	InvolvedObjectFieldPath Nullable(String),
+	InvolvedObjectLabels Map(String, String),
+	InvolvedObjectAnnotations Map(String, String),
+	InvolvedObjectDeleted Bool,
+	OwnerReferenceAPIVersions Array(LowCardinality(String)),
+	OwnerReferenceKinds Array(LowCardinality(String)),
+	OwnerReferenceNames Array(LowCardinality(String)),
+	OwnerReferenceUIDs Array(UUID),
+	ReportingComponent LowCardinality(Nullable(String)),
+	ReportingInstance Nullable(String),
+) ENGINE = MergeTree
+ORDER BY (LastTimestamp, InvolvedObjectNamespace, InvolvedObjectName)
+SETTINGS allow_nullable_key=true
+;`
+	insertStatementFmt = `INSERT INTO %s (
```

Comment:
let's use async inserts as we're making unbatched inserts

Reply:
I'm setting it in the context: https://github.com/ClickHouse/kubernetes-event-exporter/pull/2/files#diff-47af19dd403d1ce49098e087e5b7f4e64bba34e7283fe095b7ddd5706b6dbd9dR192
Is there a way to do this on the statement level? I got the context variable from our [own examples](https://github.com/ClickHouse/clickhouse-go/blob/v2.24.0/examples/std/async.go)

Reply:
missed that, looks good to me!